packages/segmented-control/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,314 @@
+  [key in Size]: string;
+} = {
+  small: css`
+    --segment-gap: 1px;
+    --padding: 0px;
+    --radius: 4px;
+  `,
+  default: css`
+    --segment-gap: 5px;
+    --padding: 3px;
+    --radius: 6px;
+  `,
+  large: css`
+    --segment-gap: 5px;
+    --padding: 3px;
+    --radius: 6px;
+  `,
+};
+
+const frameStyleFromMode: {
+  [key in Mode]: string;
+} = {
+  light: css`
+    --background-color: ${uiColors.gray.light3};
+    --border-color: transparent;
+    --border-width: 0px;
+    --inner-shadow: 0px 1px 2px rgba(0, 0, 0, 0.3) inset;
+    --outer-shadow: 0px 1px 1px #e7eeec;
+  `,
+  dark: css`
+    --background-color: ${uiColors.gray.dark3};
+    --border-color: ${uiColors.gray.dark1};
+    --border-width: 1px;
+    --inner-shadow: unset;
+    --outer-shadow: unset;
+  `,
+};
+
+const frameStyleBase = css`
+  position: relative;
+  display: grid;
+  padding: var(--padding);
+  grid-auto-flow: column;
+  grid-auto-columns: 1fr;
+  gap: var(--segment-gap);
+  align-items: center;
+  background-color: var(--background-color);
+  border-radius: var(--radius);
+  border-width: var(--border-width);
+  border-style: solid;
+  border-color: var(--border-color);
+
+  &:after {
+    position: absolute;
+    content: '';
+    width: 100%;
+    height: 100%;
+    border-radius: var(--radius);
+    box-shadow: var(--inner-shadow), var(--outer-shadow);
+    z-index: 2;
+    pointer-events: none;
+  }
+`;
+
+const indicatorStyleFromSize: {
+  [key in Size]: string;
+} = {
+  small: css`
+    --indicator-height: 100%;
+  `,
+  default: css`
+    --indicator-height: calc(100% - 2 * var(--padding));
+  `,
+  large: css`
+    --indicator-height: calc(100% - 2 * var(--padding));
+  `,
+};
+
+const indicatorStyleFromMode: {
+  [key in Mode]: string;
+} = {
+  light: css`
+    --indicator-background-color: ${uiColors.gray.light2};
+    --indicator-border-color: ${selectionBorderColor};
+  `,
+  dark: css`
+    --indicator-background-color: ${uiColors.gray.dark1};
+    --indicator-border-color: ${uiColors.gray.base};
+  `,
+};
+
+const selectionIndicatorBase = css`
+  position: absolute;
+  height: var(--indicator-height);
+  /* TODO - fix the shadow overhang over the frame */
+  box-shadow: 0px 1px 2px rgba(6, 22, 33, 0.3);
+  transition: transform 150ms ease-in-out;
+  z-index: 0;
+  border-radius: 4px;
+  border-width: 1px;
+  border-style: solid;
+  background-color: var(--indicator-background-color);
+  border-color: var(--indicator-border-color);
+`;
+
+const interactionRingStyle = css`
+  z-index: 3;
+`;
+
+/**
+ * Types
+ */
+export interface SegmentedControlProps {
+  children: React.ReactNode;
+  name?: string;
+  size?: Size;
+  darkMode?: boolean;
+  defaultValue?: string;
+  value?: string;
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  className?: string;
+  label?: string;
+}
+
+/**
+ * Component
+ */
+const SegmentedControl = React.forwardRef<
+  HTMLDivElement,
+  SegmentedControlProps
+>(function SegmentedControl(
```

Comment:
I've seen both patterns in the codebase (named functions are used in `Button` and `MenuButton`). The benefit of a named function is that any  runtime errors in the component will have the function name in the stack trace as opposed to 'anonymous'

Reply:
I see the benefit, but I'd like to see us explore if there's a better if we intend to do this going forward. Is there a better pattern we could use here to avoid the awkwardness of this?

```js
const SegmentedControl = React.forwardRef(function SegmentedControl() {})
```

Maybe we could do something along these lines (with probably a bit of typescript work as well):

```js
function SegmentedControl() {}

SegmentedControl.displayName = "SegmentedControl";

export default React.forwardRef(SegmentedControl);
```